Server/App/api/deck.py

```diff
@@ -190,12 +192,105 @@ def put(self, name, public, tags, actions, parent_deck_version, parent_user_data
         return self.make_response(response=json.dumps(deck, default=serializer),
             status=StatusCode.OK)
 
+    @authorize_request_and_create_db_connector
+    @extract_user_id
+    @require_params('name', 'public', 'tags', 'cards', 'device')
+    def post(self, name, public, tags, cards, device,
+            user_id, connector, uuid):
+        """Overwrite an existing deck."""
+
+        # Ensure this user actually owns the deck.
+        results = connector.call_procedure('FETCH_DECK', user_id, uuid)
+        if len(results) != 1:
+            return self.make_response(status=StatusCode.INTERNAL_SERVER_ERROR)
+        if results[0]['owner'] != user_id:
+            return self.make_response(status=StatusCode.FORBIDDEN)
+
+        # Make sure the deck version are consistent.
+        versions = connector.call_procedure('GET_VERSIONS', user_id, uuid)
+        if len(versions) != 1:
+            return self.make_response(status=StatusCode.NOT_FOUND)
+
+        # Begin deck edits
+        try:
+            connector.begin_transaction()
+
+            # Collect and validate metadata updates.
+            metadata_edits = {}
+            if name is not None:
```

Comment:
Since this is required, you can remove this line and the `isinstance` check should catch a none type